processor/probabilisticsamplerprocessor/logprobabilisticsampler_test.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func TestNewLogsProcessor(t *testing.T) {
+	tests := []struct {
+		name         string
+		nextConsumer consumer.Logs
+		cfg          *Config
+		wantErr      bool
+	}{
+		{
+			name: "nil_nextConsumer",
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 15.5,
+			},
+			wantErr: true,
+		},
+		{
+			name:         "happy_path",
+			nextConsumer: consumertest.NewNop(),
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 15.5,
+			},
+		},
+		{
+			name:         "happy_path_hash_seed",
+			nextConsumer: consumertest.NewNop(),
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 13.33,
+				HashSeed:           4321,
+			},
+		},
+		{
+			name:         "with_severity",
+			nextConsumer: consumertest.NewNop(),
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 13.33,
+				HashSeed:           4321,
+				Severity: []severityPair{
+					{"error", 90},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newLogsProcessor(tt.nextConsumer, tt.cfg)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func TestLogsSampling(t *testing.T) {
+	tests := []struct {
+		name     string
+		cfg      *Config
+		received int
+	}{
+		{
+			name: "happy_path",
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 100,
+			},
+			received: 2,
+		},
+		{
+			name: "nothing",
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 0,
+			},
+			received: 0,
+		},
+		{
+			name: "half",
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 49,
+			},
+			received: 1,
+		},
+		{
+			name: "nothing_except_errors",
+			cfg: &Config{
+				ProcessorSettings:  config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				SamplingPercentage: 0,
+				Severity: []severityPair{
+					{"error", 100},
+				},
+			},
+			received: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sink := new(consumertest.LogsSink)
+			processor, err := newLogsProcessor(sink, tt.cfg)
+			require.NoError(t, err)
+			logs := pdata.NewLogs()
+			lr := logs.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords()
+			for i := 0; i < 5; i++ {
```

Comment:
I'm missing something here: why 5?

Reply:
I picked a number to have enough samples to ensure the sampling worked well for the "half" case, which felt really handwavey with just 2 records. Bonus points for testing this with a bit more scale and checking that the sampling function results are consistent.